src/realm/sync/noinst/client_impl_base.hpp

```diff
@@ -1133,6 +1133,10 @@ class ClientImpl::Session {
     // the detection of download completion.
     request_ident_type m_last_triggering_download_mark = 0;
 
+    // Keep track of the DownloadBatchState for the last download message to help
+    // determine if the current LastInBatch came after a message with MoreToCome.
+    DownloadBatchState m_last_download_batch_state = DownloadBatchState::SteadyState;
```

Comment:
I guess this is to distinguish between a LastInBatch meant as last message in a bootstrap and the one meant as SteadyState right? Can't the server send a third value for SteadyState? this would help with the one message bootstrap too and also simplify all the checks we need to do on the client.

Reply:
I wish they would do that, too - but they rejected the request when I asked towards the beginning of the project :(

Reply:
I don't recall that coming up, or any specific reason that would have been unreasonable - do you remember any more context around that convo?
Don't know if it makes sense at this point in the project, but we probably could add this on the server (unless I'm overlooking a concrete reason why we shouldn't) if it would provide a significant improvement here.